### PR TITLE
Add docs CI build script

### PR DESCRIPTION
**Pull request by charlesbluca**
Adds a docs build script for gpuCI to execute on merges; addresses #579.

Depends on rapidsai/integration#250

cc @quasiben

### CODE REVIEW

File: ci/docs/build.sh
Comment (charlesbluca) [inline ci/docs/build.sh:28]: Are there any packages we would need to add to this environment to get docs builds working?

https://github.com/rapidsai/dask-cuda/blob/d39fff9cd6f766dc03cb9f6858b19caf7d3cc438/conda/environments/builddocs_py37.yml#L5-L9

File: ci/docs/build.sh
Comment (charlesbluca) [inline ci/docs/build.sh:17]: `HOME` and `NIGHTLY_VERSION` are exported but unused (copied this from cuDF's build script); are these required by gpuCI or can I remove these?

**Review by charlesbluca — COMMENTED** [review]
Some questions:

File: ci/docs/build.sh
Comment (dillon-cullinan) [reply ci/docs/build.sh:17]: HOME is a variable needed by Jenkins, and NIGHTLY_VERSION is used in the build outside this script.

File: ci/docs/build.sh
Comment (dillon-cullinan) [reply ci/docs/build.sh:27]: The docs are built in the `devel` images and automatically download some basic stuff such as rapids-doc-env. I don't think `sphinx-click` is included in our `devel` environment so this may need to be added manually.

**Review by dillon-cullinan — APPROVED** [review]
This looks fine, but read my comments above to make sure you're not missing anything, as I'm not sure if they're required or not.

File: ci/docs/build.sh
Comment (charlesbluca) [reply ci/docs/build.sh:27]: Would this need to be added from the script, or could it be added to the `devel` environment permanently?

**Comment by charlesbluca** [comment]
Waiting on rapidsai/integration#250 before we merge this.

**Comment by charlesbluca** [comment]
With rapidsai/integration#250 in, this should be good to go - might need to wait on the `devel` image to be rebuilt with the `rapids-doc-env` but after that things should work smoothly.

**Comment by charlesbluca** [comment]
@gpucibot merge

**Pull request merged**
